builder/simplify_exprs.cc

```diff
@@ -938,7 +947,9 @@ expr simplify(const class select* op, expr c, expr t, expr f) {
       
       r.rewrite(select(x, select(y, z, w), select(y, u, w)), select(y, select(x, z, u), w)) ||
       r.rewrite(select(x, select(y, z, w), select(y, z, u)), select(y, z, select(x, w, u))) ||
-
+      r.rewrite(select(x, 0, 1), x == 0) ||
+      r.rewrite(select(x, 1, 0), x != 0) ||
+      r.rewrite(select(x, y, true), y || !x, is_logical(y)) ||
```

Comment:
Aren't there 3 other similar rules we should have here, for false, and the same for the true_value case?

Reply:
I added three more rules for true_value case and true, but it sounds like you meant there should be six more rules?

Reply:
This is what I meant, looks good!